packages/next-on-pages/src/api/getRequestContext.ts

```diff
@@ -66,3 +86,36 @@ export function getRequestContext<
 
 	return cloudflareRequestContext;
 }
+
+/**
+ * detects whether the current code is running inside the (local) edge runtime or not
+ *
+ * @returns a boolean indicating whether the code is running inside the edge runtime if that can be accurately detected,
+ * 			'maybe' if the code is likely being run inside the edge runtime but we're not 100% sure of that
+ */
+function isInsideEdgeRuntime(): boolean | 'maybe' {
+	try {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		process.cwd();
+	} catch (error) {
+		if (error instanceof Error) {
+			// When certain Node.js APIs are being used the Next dev server errors with a message such as:
+			//    "A Node.js API is used (...) which is not supported in the Edge Runtime"
+			// we can rely on such error in order to discern accurately if we are or not in the edge runtime
+			// (source: https://github.com/vercel/next.js/blob/0fe68736/packages/next/src/server/web/sandbox/context.ts#L118-L122)
+			const notSupportedInEdgeError = error.message.includes(
+				'is not supported in the Edge Runtime',
+			);
+			if (!notSupportedInEdgeError) {
+				return true;
+			}
+		}
+		// an error was thrown... likely we are in the edge runtime, but there's no real guarantee
+		return 'maybe';
+	}
+
+	// for the time being we can safely assume that if `process.cwd()` doesn't error
```

Comment:
Or that they have included some polyfill?
Can we check something else like `process.release.name`?

https://nodejs.org/docs/latest/api/process.html#processrelease

Reply:
`process.release.name` sounds like a great idea! 😃 

I didn't know about it (nor noticed it when I was scanning `process` to see what we could use from that)

I'll have a look thanks! 😄

Reply:
works like a charm! 🤩👍